lib/instrumentation/modules/mongodb.js

```diff
@@ -175,13 +173,49 @@ module.exports = (mongodb, agent, { version, enabled }) => {
     }
   }
 
-  function onEnd(event) {
+  function onSuccess(event) {
+    // `event` is a `CommandSucceededEvent`
+    // https://github.com/mongodb/specifications/blob/master/source/command-logging-and-monitoring/command-logging-and-monitoring.rst#command-succeeded-message
+    // E.g. with mongodb@6.2.0:
+    //   CommandSucceededEvent {
+    //     connectionId: '127.0.0.1:27017',
+    //     requestId: 1,
+    //     duration: 1,
+    //     reply: { ... }
+    //   }
     if (!activeSpans.has(event.requestId)) return;
-    const { failure, reply } = event;
-    const hasFailed = failure || (reply && reply.writeErrors);
+
+    const span = activeSpans.get(event.requestId);
+    activeSpans.delete(event.requestId);
+
+    // From mongodb@3.6.0 and up some commands like `deleteOne` may contain
+    // error data inside the `reply` property. It makes sense to capture it.
+    const writeErrors = event && event.reply && event.reply.writeErrors;
+
+    if (writeErrors && writeErrors.length) {
+      // TODO: this gets linked to the parent transaction. How to link it to the span?
```

Comment:
`captureError` has a `parent` option for just this case:

```
//   - `opts.parent` - A Transaction or Span instance to make the parent of
//     this error. If not given (undefined), then the current span or
//     transaction will be used. If `null` is given, then no span or transaction
//     will be used.
```

I guess the emitted `CommandSucceededEvent`s (and the other events) are not in the context of the created span, which is fine.

Reply:
done!